core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2570,10 +2570,23 @@ class SparkContext(config: SparkConf) extends Logging {
 
   private[spark] def newShuffleId(): Int = nextShuffleId.getAndIncrement()
 
-  private val nextRddId = new AtomicInteger(0)
+  private var nextRddId = new AtomicInteger(0)
 
   /** Register a new RDD, returning its RDD ID */
-  private[spark] def newRddId(): Int = nextRddId.getAndIncrement()
+  private[spark] def newRddId(): Int = {
+    var id = nextRddId.getAndIncrement()
+    if (id >= 0) {
+        return id
+    }
+    this.synchronized {
+      id = nextRddId.getAndIncrement()
+      if (id < 0) {
+        nextRddId = new AtomicInteger(0)
+        id = nextRddId.getAndIncrement()
```

Comment:
Can we use this?
``` 
nextRddId = new AtomicInteger(1)
id = nextRddId 
```

Reply:
Thank you for your review. I updated the code.